add runninng Autoware without CUDA (#128)
autowarefoundation/autoware-documentation: docs/how-to-guides/running-autoware-without-cuda.md

# Running Autoware without CUDA

Although CUDA installation is recommended to achieve better performance for object detection and traffic light recognition in Autoware Universe, it is possible to run these algorithms without CUDA.
The following subsections briefly explain how to run each algorithm in such an environment.

## Running 2D/3D object detection without CUDA

> The following three object detection modules require CUDA:
> 
> - `lidar_centerpoint`
> - `lidar_apollo_instance_segmentation`
> - `tensorrt_yolo`
> 
> Instead of these modules, you can use the `euclidean_cluster` module which contains a 3D object detection algorithm that does not require CUDA. For more details, refer to the [`euclidean_cluster` module's README file](https://github.com/autowarefoundation/autoware.universe/tree/main/perception/euclidean_cluster).

Reviewer: @kminoda I think it is useful to know the different possible configurations so I have included them in my suggestion below. Alternatively, would it be better to include the information about possible configurations in the design documentation for the Perception component?

```suggestion
Autoware Universe's object detection can be run using one of five possible configurations:

- `lidar_centerpoint`
- `lidar_apollo_instance_segmentation`
- `lidar-apollo` + `tensorrt_yolo`
- `lidar-centerpoint` + `tensorrt_yolo`
- `euclidean_cluster`

Of these five configurations, only the last one (`euclidean_cluster`) can be run without CUDA. For more details, refer to the [`euclidean_cluster` module's README file](https://github.com/autowarefoundation/autoware.universe/tree/main/perception/euclidean_cluster).
```

> The following three object detection modules require CUDA:
> 
> - `lidar_centerpoint`
> - `lidar_apollo_instance_segmentation`
> - `tensorrt_yolo`
> 
> Instead of these modules, you can use the `euclidean_cluster` module which contains a 3D object detection algorithm that does not require CUDA. For more details, refer to the [`euclidean_cluster` module's README file](https://github.com/autowarefoundation/autoware.universe/tree/main/perception/euclidean_cluster).

Reviewer: @kminoda I think it is useful to know the different possible configurations so I have included them in my suggestion below. Alternatively, would it be better to include the information about possible configurations in the design documentation for the Perception component?

```suggestion
Autoware Universe's object detection can be run using one of five possible configurations:

- `lidar_centerpoint`
- `lidar_apollo_instance_segmentation`
- `lidar-apollo` + `tensorrt_yolo`
- `lidar-centerpoint` + `tensorrt_yolo`
- `euclidean_cluster`

Of these five configurations, only the last one (`euclidean_cluster`) can be run without CUDA. For more details, refer to the [`euclidean_cluster` module's README file](https://github.com/autowarefoundation/autoware.universe/tree/main/perception/euclidean_cluster).
```

Author: I like your suggestion. Thanks!
I reflected the suggestion as below:
https://github.com/autowarefoundation/autoware-documentation/pull/128/commits/876081ae5df71cf1fc13e761a9a8c6fa80569a17

# Running Autoware without CUDA

> Two following algorithms in Autoware depend on a CUDA environment.

Reviewer 2: Since Autoware Core won't depend on CUDA, I think it's good to say Autoware Universe here.

```suggestion
Two following algorithms in Autoware Universe depend on a CUDA environment.
```

Author: https://github.com/autowarefoundation/autoware-documentation/pull/128/commits/ad7a25824788219877f9b9ffde001e6e0520ab41
Fixed! Thank you for the suggestion

# Running Autoware without CUDA

> Two following algorithms in Autoware Universe depend on a CUDA environment.
> 
> - 2D/3D object detection
> - traffic light detection
> 
> However, it is still possible to run Autoware in the environment without CUDA.
> The following subsections briefly explain how to run these algorithms in such an environment.
> 
> !!! warning
> 
>     It is highly recommended to prepare a CUDA environment to achieve better performance.

Reviewer: I think the opening section can be simplified as follows:

```suggestion
Although CUDA installation is recommended to achieve better performance for object and traffic light detection in Autoware Universe, it is possible to run these algorithms without CUDA. 

The following subsections briefly explain how to run each algorithm in such an environment.
```

## Running 2D/3D object detection without CUDA

> As for the object detection system in Autoware, the following three packages require the CUDA environment:

Reviewer: Rewording suggestion.

```suggestion
The following three object detection modules require CUDA:
```

- `lidar_centerpoint`
- `lidar_apollo_instance_segmentation`
- `tensorrt_yolo`

> Instead of these modules, you can use the `euclidean_cluster` module, which is a 3D object detection algorithm that does not require a CUDA environment (refer to [the readme](https://github.com/autowarefoundation/autoware.universe/tree/main/perception/euclidean_cluster) for detail).

Reviewer: Reworded/rephrased. Is it correct to say that the two lidar* modules related to LiDAR-based detection whilst `tensorrt_yolo` is used for camera-based detection? If so, then is it also correct to say that `euclidean_cluster` can onyl be used instead of the lidar-related modules?

```suggestion
Instead of these modules, you can use the `euclidean_cluster` module which contains a 3D object detection algorithm that does not require CUDA. For more details, refer to the [`euclidean_cluster` module's README file](https://github.com/autowarefoundation/autoware.universe/tree/main/perception/euclidean_cluster).
```

## Running traffic light detection without CUDA

> As for the traffic light recognition system in Autoware, there are two package that require the CUDA environment:

Reviewer: Rewording suggestion.

```suggestion
For traffic light recognition (both detection and classification), there are two modules that require CUDA:
```

Author: Thank you for the suggestion!
I have reflected the comments at here https://github.com/autowarefoundation/autoware-documentation/pull/128/commits/2f9ea059376f62b1fc11ba14173261e74aac19fb

- `traffic_light_ssd_fine_detector`
- `traffic_light_classifier`

> To run traffic light detection without CUDA, set [`enable_fine_detection` in this file](https://github.com/autowarefoundation/autoware.universe/blob/9445f3a7acd645d12a64507c3d3bfa57e74a3634/launch/tier4_perception_launch/launch/traffic_light_recognition/traffic_light.launch.xml#L3) to `false` so that `traffic_light_ssd_fine_detector` be disabled and the traffic light detection solely be executed by `map_based_traffic_light_detector`.

Author: Thank you for the suggestion!
I have reflected the comments at here https://github.com/autowarefoundation/autoware-documentation/pull/128/commits/2f9ea059376f62b1fc11ba14173261e74aac19fb

> To run traffic light classification without CUDA, set [`use_gpu` in this file](https://github.com/autowarefoundation/autoware.universe/blob/9445f3a7acd645d12a64507c3d3bfa57e74a3634/perception/traffic_light_classifier/launch/traffic_light_classifier.launch.xml#L7) to `false` so that `traffic_light_classifier` use classification algorithm that does not use CUDA environment or GPU.

Author: Thank you for the suggestion!
I have reflected the comments at here https://github.com/autowarefoundation/autoware-documentation/pull/128/commits/2f9ea059376f62b1fc11ba14173261e74aac19fb
